In [1]:
import numpy as np
import pandas as pd

from sklearn.svm import SVC
from sklearn.metrics import accuracy_score, confusion_matrix
from sklearn.preprocessing import LabelEncoder, StandardScaler

In [2]:
train = pd.read_csv('train.csv')
train.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [3]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
PassengerId    891 non-null int64
Survived       891 non-null int64
Pclass         891 non-null int64
Name           891 non-null object
Sex            891 non-null object
Age            714 non-null float64
SibSp          891 non-null int64
Parch          891 non-null int64
Ticket         891 non-null object
Fare           891 non-null float64
Cabin          204 non-null object
Embarked       889 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 83.6+ KB


In [4]:
train.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [5]:
train.drop(['Cabin'],axis=1,inplace=True)

In [6]:
train['Age'].fillna(value=train.Age.median(),inplace=True)

In [7]:
train.dropna(inplace=True)

In [8]:
train.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

In [9]:
train['Age'][train['Age']<1]

78     0.83
305    0.92
469    0.75
644    0.75
755    0.67
803    0.42
831    0.83
Name: Age, dtype: float64

In [10]:
train['Age'][train['Age']<1] = 1

C:\Users\Dell\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [11]:
train['Age'] = train['Age'].astype(int)

In [12]:
train.set_index('PassengerId',inplace=True)

In [13]:
train.shape

(889, 10)

In [14]:
test = pd.read_csv('test.csv')
test.head()

PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
0  34.5      0      0   330911   7.8292   NaN        Q  
1  47.0      1      0   363272   7.0000   NaN        S  
2  62.0      0      0   240276   9.6875   NaN        Q  
3  27.0      0      0   315154   8.6625   NaN        S  
4  22.0      1      1  3101298  12.2875   NaN        S

In [15]:
test.set_index('PassengerId',inplace=True)

In [16]:
test1 = pd.read_csv('gender_submission.csv')

In [17]:
test1.set_index('PassengerId',inplace=True)

In [18]:
test = pd.concat([test,test1],axis=1)

In [19]:
test.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 418 entries, 892 to 1309
Data columns (total 11 columns):
Pclass      418 non-null int64
Name        418 non-null object
Sex         418 non-null object
Age         332 non-null float64
SibSp       418 non-null int64
Parch       418 non-null int64
Ticket      418 non-null object
Fare        417 non-null float64
Cabin       91 non-null object
Embarked    418 non-null object
Survived    418 non-null int64
dtypes: float64(2), int64(4), object(5)
memory usage: 39.2+ KB


In [20]:
test.isnull().sum()

Pclass        0
Name          0
Sex           0
Age          86
SibSp         0
Parch         0
Ticket        0
Fare          1
Cabin       327
Embarked      0
Survived      0
dtype: int64

In [21]:
test.drop(['Cabin'],axis=1,inplace=True)

In [22]:
test['Age'].fillna(value=test.Age.median(),inplace=True)

In [23]:
test.isnull().sum()

Pclass      0
Name        0
Sex         0
Age         0
SibSp       0
Parch       0
Ticket      0
Fare        1
Embarked    0
Survived    0
dtype: int64

In [24]:
test.dropna(inplace=True)

In [25]:
test.isnull().sum()

Pclass      0
Name        0
Sex         0
Age         0
SibSp       0
Parch       0
Ticket      0
Fare        0
Embarked    0
Survived    0
dtype: int64

In [26]:
test.shape

(417, 10)

In [27]:
test['Age'][test['Age']<1]

PassengerId
1093    0.33
1142    0.92
1173    0.75
1199    0.83
1246    0.17
Name: Age, dtype: float64

In [28]:
test['Age'][test['Age']<1] = 1

C:\Users\Dell\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [29]:
test['Age'] = test['Age'].astype(int)

In [30]:
test.shape

(417, 10)

In [31]:
le = LabelEncoder()
train.Sex = le.fit_transform(train.Sex)
le.classes_

array(['female', 'male'], dtype=object)

In [32]:
test.Sex = le.transform(test.Sex)
le.classes_

array(['female', 'male'], dtype=object)

In [33]:
X_train = train[['Pclass','Sex','Age']]
X_test = test[['Pclass','Sex','Age']]
Y_train = train['Survived']
Y_test = test['Survived']

In [34]:
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

In [35]:
classifier = SVC(kernel='poly', degree=5, random_state=0)
classifier.fit(X_train,Y_train)

SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
  decision_function_shape='ovr', degree=5, gamma='auto', kernel='poly',
  max_iter=-1, probability=False, random_state=0, shrinking=True,
  tol=0.001, verbose=False)

In [36]:
y_pred = classifier.predict(X_test)
accuracy_score(Y_test,y_pred)

0.9856115107913669

In [37]:
confusion_matrix(Y_test,y_pred)

array([[262,   3],
       [  3, 149]], dtype=int64)

In [38]:
pred1 = classifier.predict(sc.transform(np.array([[1,1,20]])))
pred1

C:\Users\Dell\Anaconda3\lib\site-packages\sklearn\utils\validation.py:475: DataConversionWarning: Data with input dtype int32 was converted to float64 by StandardScaler.
  warnings.warn(msg, DataConversionWarning)


array([0], dtype=int64)

In [40]:
pred2 = classifier.predict(sc.transform(np.array([[1,0,20]])))
pred2

C:\Users\Dell\Anaconda3\lib\site-packages\sklearn\utils\validation.py:475: DataConversionWarning: Data with input dtype int32 was converted to float64 by StandardScaler.
  warnings.warn(msg, DataConversionWarning)


array([1], dtype=int64)

In [41]:
pred3 = classifier.predict(sc.transform(np.array([[2,1,20]])))
pred3

C:\Users\Dell\Anaconda3\lib\site-packages\sklearn\utils\validation.py:475: DataConversionWarning: Data with input dtype int32 was converted to float64 by StandardScaler.
  warnings.warn(msg, DataConversionWarning)


array([0], dtype=int64)

In [42]:
pred4 = classifier.predict(sc.transform(np.array([[2,0,20]])))
pred4

C:\Users\Dell\Anaconda3\lib\site-packages\sklearn\utils\validation.py:475: DataConversionWarning: Data with input dtype int32 was converted to float64 by StandardScaler.
  warnings.warn(msg, DataConversionWarning)


array([1], dtype=int64)

In [43]:
pred5 = classifier.predict(sc.transform(np.array([[3,1,20]])))
pred5

C:\Users\Dell\Anaconda3\lib\site-packages\sklearn\utils\validation.py:475: DataConversionWarning: Data with input dtype int32 was converted to float64 by StandardScaler.
  warnings.warn(msg, DataConversionWarning)


array([0], dtype=int64)

In [44]:
pred6 = classifier.predict(sc.transform(np.array([[3,0,20]])))
pred6

C:\Users\Dell\Anaconda3\lib\site-packages\sklearn\utils\validation.py:475: DataConversionWarning: Data with input dtype int32 was converted to float64 by StandardScaler.
  warnings.warn(msg, DataConversionWarning)


array([1], dtype=int64)

In [84]:
pred7 = classifier.predict(sc.transform(np.array([[3,1,10]])))
pred7

C:\Users\Dell\Anaconda3\lib\site-packages\sklearn\utils\validation.py:475: DataConversionWarning: Data with input dtype int32 was converted to float64 by StandardScaler.
  warnings.warn(msg, DataConversionWarning)


array([0], dtype=int64)

### ------------------------------------------------------------------------------------
### If the passenger was belong to first class then there is 63% probability that passenger was survived, if passenger was from Second class then there is 47% chance that passenger was survived but if passenger was from Third class then there is only 24% chance that passenger was survived 
### If the passenger is Female then there is 74% probability that she was survived but if passenger is Male then there is only 19% probability that he is survived
### If the passenger Age is less than 15yr then there is 60% probability that the passenger was survived otherwise only 36% probability that passenger was survived